src/plugins/kibana_utils/public/state_management/url/kbn_url_storage.ts

```diff
@@ -154,7 +154,7 @@ export const createKbnUrlControls = (
   let shouldReplace = true;
 
   function updateUrl(newUrl: string, replace = false): string | undefined {
-    const currentUrl = getCurrentUrl();
+    const currentUrl = getCurrentUrl(history);
```

Comment:
This is the main change. Now when updating the url instead of using `window.location.href`, we are using current history instance.